charts/gardener/provider-local/values.yaml

```diff
@@ -2,6 +2,7 @@ image: local-skaffold/gardener-extension-provider-local:latest
 imagePullPolicy: IfNotPresent
 
 replicaCount: 1
+priorityClassName: gardener-system-900
```

Comment:
The priority class is not created by this helm chart, right? So the assumption is that the priority class is already present in the system or did I miss something?

Reply:
The priority class is created by some "gardener component" as described in https://github.com/gardener/gardener/blob/master/docs/development/priority-classes.md . The extensions typically wait to run in a seed and use one of the PCs [here](https://github.com/gardener/gardener/blob/master/docs/development/priority-classes.md#priorityclasses-for-seed-system-components) which are created by [gardenlet](https://github.com/gardener/gardener/blob/b55d4f13c9a08599d6569b19daf01c49c1dbd436/pkg/component/seed/system/system.go#L157). 

Now for the usecase of the gardener operator, we don't have gardenlet running so the operator must install some PCs for its components. Though the extension having the PC hardcoded is not helping this scenario unless the operator installs all the PCs expected in a seed.

But I find your question more than fair and made me reconsider the solution. The PR of course is made after the hackathon results, but the solution of the hackathon is probably not optimal. Making the PC "configurable" here in such a way, means that the PC is now something that should be passed somehow from the chart's values => making it something that the gardenlet or operator must do for all the extensions. That doesn't really sound nice.

Instead a "better" final solution, would be that the chart is aware of an "environment" (something we used for the provider-local in the hackathon and most likely will be used in the final product too). Then in the chart we can have a conditional to check if we are running in "operator" or "seed" mode and let the extension pick one of the expected PCs for that environment, just like how it is documented for the seeds now. WDYT ?

Reply:
To add, if you think the proposal in my comment above is more reasonable it makes sense to close this PR and will be reintroduced with a better chart along with the controller.

The clusterrolebinding change, we need 100% but it's so small that can be combined with another PR.

Reply:
Both approaches work for me. You can continue with this PR and adapt it as required or close/open a new one. I leave it up to you.